src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorProjectEngine.cs

```diff
@@ -241,6 +241,11 @@ private static void AddComponentFeatures(RazorProjectEngineBuilder builder, Razo
             ComponentPreserveWhitespaceDirective.Register(builder);
         }
 
+        if (razorLanguageVersion.CompareTo(RazorLanguageVersion.Version_8_0) >= 0)
```

Comment:
Not for this PR, more of a meta question, but for features like this where runtime API detection would work 100%, should we prefer it over RazorLanguageVersion? Seems a bit more "source-generatey" :)

Reply:
That is a good question, and something I'm not sure we've decided how to handle yet. One of the reasons on the csharp side for having language version is so that different developers can span different VS versions. 

While we can do feature detection and just opt-in, it removes the ability to 'lock' the language to a specific version. I think I'll start a thread about this and get some opinions.

Reply:
Feels like RazorLanguageVersion should be used, so one can fix it to 7 even when using .net 8 - for example to avoid the break described in https://github.com/dotnet/Razor-Language-Design/discussions/3#discussioncomment-6662874